### PR TITLE
makefile(all): add go-generate target

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/7603

### What is changed and how it works?

* Remove scripts/generate-msgp-code.sh.
* Replace `generate-msgp-code` with `go-generate`.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — APPROVED** [review]
Looks good to me. Thanks!

File: cdc/redo/writer/mock_RedoLogWriter.go [hunk 1]
```diff
@@ -1,17 +1,4 @@
-//  Copyright 2021 PingCAP, Inc.
```
Comment (zhaoxinyu) [inline cdc/redo/writer/mock_RedoLogWriter.go:1]: Should the copyright information be retained?

**Review by zhaoxinyu — APPROVED** [review]

File: cdc/redo/writer/mock_RedoLogWriter.go (hunk 1 above)
Comment (gozssky) [reply cdc/redo/writer/mock_RedoLogWriter.go:1]: This is a fully generated file like pb.go. So I think it is unnecessary.

**Pull request merged**